scripts/react_native_pods.rb

```diff
@@ -673,18 +673,18 @@ def downloadAndConfigureHermesSource(react_native_path)
     Pod::UI.puts "[Hermes] Downloading Hermes source code (#{hermes_tarball_url})"
     system("curl #{hermes_tarball_url} -Lo #{hermes_tarball_path}")
   end
-  Pod::UI.puts "[Hermes] Extracting Hermes (#{hermes_tag_sha})"
 
+  
   hermesc_macos_path = "#{sdks_dir}/hermesc/macos/build_host_hermesc"
   hermesc_macos_link = "#{hermes_dir}/utils/build_host_hermesc"
   if (File.exist?(hermesc_macos_path))
     # If hermesc is present, create a symbolic link in the hermes source directory to avoid re-building hermesc
     Pod::UI.puts "[Hermes] Using pre-compiled Hermes Compiler from #{hermesc_macos_path}"
     system("ln -s #{hermesc_macos_path} #{hermesc_macos_link}")
   end
-
-  # TODO: Integrate this temporary hermes-engine.podspec into the actual one located in facebook/hermes
-  system("cp #{sdks_dir}/hermes-engine.podspec #{hermes_dir}/hermes-engine.podspec")
+  
+  Pod::UI.puts "[Hermes] Extracting Hermes (#{hermes_tag_sha})"
+  system("tar -zxf #{hermes_tarball_path} --strip-components=1 --directory #{hermes_dir}")
```

Comment:
Please add:

```
system("tar -zxf #{hermes_tarball_path} --strip-components=1 --directory #{hermes_dir}")
``` 

at line 676.

And don't remove the copy of the `.podspec` file.

Reply:
Fixed under [00570d9](https://github.com/facebook/react-native/pull/33585/commits/00570d94d2cb4e86609091ac61d12ea03a81887d).